pkg/lint/lintersdb/manager.go

```diff
@@ -939,32 +143,143 @@ func (m *Manager) GetAllEnabledByDefaultLinters() []*linter.Config {
+	if m.cfg.Linters.Fast {
+		for name, lc := range resultLintersSet {
+			if lc.IsSlowLinter() {
+				delete(resultLintersSet, name)
 			}
 		}
 	}
 
-	return ret
+	for _, name := range m.cfg.Linters.Enable {
+		for _, lc := range m.GetLinterConfigs(name) {
+			// it's important to use lc.Name() nor name because name can be alias
+			resultLintersSet[lc.Name()] = lc
+		}
+	}
+
+	for _, name := range m.cfg.Linters.Disable {
+		for _, lc := range m.GetLinterConfigs(name) {
+			// it's important to use lc.Name() nor name because name can be alias
+			delete(resultLintersSet, lc.Name())
+		}
+	}
+
+	// typecheck is not a real linter and cannot be disabled.
+	if _, ok := resultLintersSet["typecheck"]; !ok && (m.cfg == nil || !m.cfg.InternalCmdTest) {
+		for _, lc := range m.GetLinterConfigs("typecheck") {
+			// it's important to use lc.Name() nor name because name can be alias
+			resultLintersSet[lc.Name()] = lc
+		}
+	}
+
+	return resultLintersSet
 }
 
-func linterConfigsToMap(lcs []*linter.Config) map[string]*linter.Config {
-	ret := map[string]*linter.Config{}
+func (m *Manager) combineGoAnalysisLinters(linters map[string]*linter.Config) {
+	var goanalysisLinters []*goanalysis.Linter
+	goanalysisPresets := map[string]bool{}
+	for _, lc := range linters {
+		lnt, ok := lc.Linter.(*goanalysis.Linter)
+		if !ok {
+			continue
+		}
+		if lnt.LoadMode() == goanalysis.LoadModeWholeProgram {
+			// It's ineffective by CPU and memory to run whole-program and incremental analyzers at once.
+			continue
+		}
+		goanalysisLinters = append(goanalysisLinters, lnt)
+		for _, p := range lc.InPresets {
+			goanalysisPresets[p] = true
+		}
+	}
+
+	if len(goanalysisLinters) <= 1 {
+		m.debugf("Didn't combine go/analysis linters: got only %d linters", len(goanalysisLinters))
+		return
+	}
+
+	for _, lnt := range goanalysisLinters {
+		delete(linters, lnt.Name())
+	}
+
+	// Make order of execution of go/analysis analyzers stable.
+	sort.Slice(goanalysisLinters, func(i, j int) bool {
+		a, b := goanalysisLinters[i], goanalysisLinters[j]
+
+		if b.Name() == linter.LastLinter {
+			return true
+		}
+
+		if a.Name() == linter.LastLinter {
+			return false
+		}
+
+		return a.Name() <= b.Name()
+	})
+
+	ml := goanalysis.NewMetaLinter(goanalysisLinters)
+
+	presets := maps.Keys(goanalysisPresets)
+	sort.Strings(presets)
+
+	mlConfig := &linter.Config{
+		Linter:           ml,
+		EnabledByDefault: false,
+		InPresets:        presets,
+		AlternativeNames: nil,
+		OriginalURL:      "",
+	}
+
+	mlConfig = mlConfig.WithLoadForGoAnalysis()
+
+	linters[ml.Name()] = mlConfig
+	m.debugf("Combined %d go/analysis linters into one metalinter", len(goanalysisLinters))
+}
+
+func (m *Manager) verbosePrintLintersStatus(lcs map[string]*linter.Config) {
+	var linterNames []string
 	for _, lc := range lcs {
-		lc := lc // local copy
-		ret[lc.Name()] = lc
+		if lc.Internal {
+			continue
+		}
+
+		linterNames = append(linterNames, lc.Name())
 	}
+	sort.StringSlice(linterNames).Sort()
```

Comment:
Perhaps we should consider a bit of refactoring?

```suggestion
	sort.Strings(linterNames)
```